Src/BootCamp.Chapter/CaesarCipher.cs

```diff
@@ -1,22 +1,242 @@
+
+        /// <summary>
+        /// Transfers the elements from one of the arrays in the bi-dimensional array to one dimension array.
+        /// </summary>
+        /// <param name="inputArray">The array from witch the transfer</param>
+        /// <param name="arrayNumber"></param>
+        /// <returns>One dimension array based on argument 0 or 1</returns>
+        private static int[] Deconstruct2DArray(int[][] inputArray, int arrayNumber)
+        {
+            bool argumentsAreNotValid = (inputArray == null || inputArray.Length == 0) && (arrayNumber < 0 || arrayNumber > 1);
+            if (argumentsAreNotValid)
+            {
+                Console.WriteLine("Arguments are not valid!");
+                return default;
+            }
+
+            var newArray = new int[inputArray[arrayNumber].Length];
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                newArray[i] = inputArray[arrayNumber][i];
+            }
+            return newArray;
+        }
+
+        /// <summary>
+        /// Implementation of Caesar Cipher character shift algorithm
+        /// </summary>
+        /// <param name="inputCharacter">character to be shifted</param>
+        /// <param name="shift">numbers of characters to be shifted</param>
+        /// <returns>resulting character after shift is performed</returns>
+        private static char EncodeCharacter(char inputCharacter, int shift)
+        {
+            if (!IsPrintableChar(inputCharacter))
+            {
+                return inputCharacter;
+            }
+
+            var output = (char)(Mod(inputCharacter + shift - BaseCharacter, TopCharacter - BaseCharacter + 1) + BaseCharacter);
+            return output;
+        }
+
+        /// <summary>
+        /// Finds possible encryption keys for Caesar Cipher based on character repetition array
+        /// </summary>
+        /// <param name="characterOccurrences">bi-dimensional array containing</param>
+        /// <returns>an array with possible decryption keys</returns>
+        private static int[] FindPossibleKeys(int[][] characterOccurrences)
+        {
+            // I could have done a characterOccurrences[0] and [1] but
+            // despite the fact It's more complicated I learned more about arrays.
+            var repeatedCharacters = Deconstruct2DArray(characterOccurrences, 0);
+            var numberOfOccurrences = Deconstruct2DArray(characterOccurrences, 1);
+            var highestOccurrence = ArrayOps.FindMaxValue(numberOfOccurrences);
+
+            var possibleKeys = Array.Empty<int>();
+            int currentKey;
+
+            for (int i = 0; i < repeatedCharacters.Length; i++)
+            {
+                if (numberOfOccurrences[i] >= highestOccurrence - KeyAccuracy)
+                {
+                    currentKey = repeatedCharacters[i] - BaseCharacter;
+                    possibleKeys = ArrayOps.InsertLast(possibleKeys, currentKey);
+                }
+            }
```

Comment:
I notice some similarities to 112ff. You could think of trying to make a general function out of both and call that instead. However it looks fine. Don't try to force it. Maybe it's "impossible". I haven't thought about it.